tests/core/validation/test_validation.py

```diff
@@ -0,0 +1,408 @@
+    (
+        (tuple(), 0, True),
+        ([1], 0, False),
+        ({'A': 'B', 'C': 1}, 3, True),
+        (set(['A', 'B', 1, 2]), 2, False),
+        ('abcde', 5, True),
+        (b'123', 3, True),
+        (range(1,10), 15, True),
+    ),
+)
+def test_validate_length_lte(value, maximum_length, is_valid):
+    if is_valid:
+        validate_length_lte(value, maximum_length)
+    else:
+        with pytest.raises(ValidationError):
+            validate_length_lte(value, maximum_length)
+
+
+@pytest.mark.parametrize(
+    "value,minimum,is_valid",
+    (
+        (1, 2, False),
+        ('a','a', True),
+        (3, 2, True),
+    ),
+)
+def test_validate_gte(value, minimum, is_valid):
+    if is_valid:
+        validate_gte(value, minimum)
+    else: 
+        with pytest.raises(ValidationError):
+            validate_gte(value, minimum)
+
+
+@pytest.mark.parametrize(
+    "value,minimum,is_valid",
+    (
+        (1, 2, False),
+        ('a','a', False),
+        (3, 2, True),
+    ),
+)
+def test_validate_gt(value, minimum, is_valid):
+    if is_valid:
+        validate_gt(value, minimum)
+    else: 
+        with pytest.raises(ValidationError):
+            validate_gt(value, minimum)
+
+
+@pytest.mark.parametrize(
+    "value,maximum,is_valid",
+    (
+        (1, 2, True),
+        ('a', 'a', True),
+        (3, 2, False),
+    ),
+)
+def test_validate_lte(value, maximum, is_valid):
+    if is_valid:
+        validate_lte(value, maximum)
+    else: 
+        with pytest.raises(ValidationError):
+            validate_lte(value, maximum)
+
+
+@pytest.mark.parametrize(
+    "value,maximum,is_valid",
+    (
+        (1, 2, True),
+        ('a', 'a', False),
+        (3, 2, False),
+    ),
+)
+def test_validate_lt(value, maximum, is_valid):
+    if is_valid:
+        validate_lt(value, maximum)
+    else: 
+        with pytest.raises(ValidationError):
+            validate_lt(value, maximum)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        ('10010010010010010010', False),
+        (b'10010010010010010010', True),
+        (b'abc', False),
+        (b'12345', False),
+    ),
+)
+def test_validate_canonical_address(value, is_valid):
+    if is_valid:
+        validate_canonical_address(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_canonical_address(value)
+
+
+@pytest.mark.parametrize(
+    "value,multiple_of,is_valid",
+    (
+        (1, 1, True),
+        (4, 2, True),
+        (773, 3, False),
+    ),
+)
+def test_validate_multiple_of(value, multiple_of, is_valid):
+    if is_valid:
+        validate_multiple_of(value, multiple_of)
+    else:
+        with pytest.raises(ValidationError):
+            validate_multiple_of(value, multiple_of) 
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        # (0, False),
+        (1, False),
+        (8, True),
+        (64, True),
+        (773, False),
+    ),
+)
+def test_validate_multiple_of_8(value, is_valid):
```

Comment:
Some quick grepping indicates this validation function is not used anywhere in the codebase.  Do you mind just removing it?